File: src/ProjectR.Scripting/content/scripts/spells/lua/Extinction.lua
function GetTargetType()
    return Enemies
end

function GetName()
  return "Extinction"
end

function GetDescription()
    return "Meteors rain down upon your enemies."
end

function IsSupportSpell()
    return false
end

function GetMPCost()
    return .52
end

function GetDelay()
  return .15
end

function GetSpellType()
  return Magical
end

function GetMasteries()
  return { FIR }
end

function SpellEffect()
  local damage = (5.0625 * aMD * (aFIR/100) - 1.125 * dMR) * (100/dFIR)
  d:TakeDamage(damage)
end

File: src/ProjectR.Scripting/content/scripts/spells/Extinction.cs
using System.Collections.Generic;
using ProjectR.Interfaces;
using ProjectR.Interfaces.Model;
using ProjectR.Interfaces.Model.Stats;

namespace ProjectR.Scripting.Spells
{
    public class Extinction : SpellScriptBase
    {
        public override string Name { get { return "Extinction"; } }
        public override string Description { get { return "Meteors rain down upon your enemies."; } }

        public override TargetType TargetType { get { return TargetType.Enemies; } }
        public override IList<EleMastery> Masteries { get { return new EleMastery[] { EleMastery.FIR, }; } }
        public override SpellType SpellType { get { return SpellType.Magical; } }
        public override bool IsSupportSpell { get { return false; } }

        public override double MPCost { get { return 52; } }
        public override double Delay { get { return .15; } }

        public override void SpellEffect(ICharacter caster, ICharacter target)
        {
            var damage = (5.0625 * cMD * (cFIR / 100) - 1.125 * tMR) * (100/tFIR);
            DealDamage(damage);
        }
    }
}